hive/src/main/java/com/netease/arctic/hive/op/OverwriteHiveFiles.java

```diff
@@ -38,26 +43,31 @@ public class OverwriteHiveFiles implements OverwriteFiles {
 
   private static final Logger LOG = LoggerFactory.getLogger(OverwriteHiveFiles.class);
 
-  final OverwriteFiles delegate;
-  final UnkeyedHiveTable table;
-  final HMSClient hmsClient;
-  final HMSClient transactionClient;
-  final String db;
-  final String tableName;
-
-  final Table hiveTable;
-
-  Expression expr;
-  List<DataFile> addFiles = Lists.newArrayList();
-  List<DataFile> deleteFiles = Lists.newArrayList();
-
-  List<Partition> partitionToDelete = Lists.newArrayList();
-  List<Partition> partitionToCreate = Lists.newArrayList();
-  long txId = -1;
-
-  public OverwriteHiveFiles(
-      OverwriteFiles delegate, UnkeyedHiveTable table, HMSClient hmsClient, HMSClient transactionClient) {
-    this.delegate = delegate;
+  private final Transaction transaction;
+  private final boolean insideTransaction;
+  private final OverwriteFiles delegate;
+  private final UnkeyedHiveTable table;
+  private final HMSClient hmsClient;
+  private final HMSClient transactionClient;
+  private final String db;
+  private final String tableName;
+
+  private final Table hiveTable;
+
+  private Expression expr;
+  private final List<DataFile> addFiles = Lists.newArrayList();
+  private final List<DataFile> deleteFiles = Lists.newArrayList();
+
+  private Map<StructLike, Partition> partitionToDelete = Maps.newHashMap();
+  private Map<StructLike, Partition> partitionToCreate = Maps.newHashMap();
+  private String unpartitionTableLocation;
+  private long txId = -1;
+
+  public OverwriteHiveFiles(Transaction transaction, boolean insideTransaction, UnkeyedHiveTable table,
```

Comment:
insideTransaction is duplicate with transaction.

Reply:
yes, i will fix it in next pr.